File: marketmotormicroservices/producto-service/src/main/java/com/jma/productoservice/ordenCompra/domain/response/OrdenCompraResponse.java
package com.jma.productoservice.ordenCompra.domain.response;


import com.jma.productoservice.ordenCompra.domain.dto.OrdenCompraDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrdenCompraResponse {

    private List<OrdenCompraDto> content;
    private int pageNo;
    private int pageSize;
    private long totalElements;
    private int totalPages;
    private boolean last;

}

File: frontend-marketmotor/src/app/models/responseapi/OrdenCompraResponse.ts
import { OrdenCompra } from "../dtos/OrdenCompra";

export class OrdenCompraResponse{
    content: OrdenCompra[];
    pageNo: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    last: boolean;

    constructor(){
        this.content  = [new OrdenCompra()]
        this.pageNo = 0;
        this.pageSize = 0;
        this.totalElements = 0;
        this.totalPages = 0;
        this.last = false;
    }
}